src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/Data.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck.transformation
+
+import org.objectweb.asm.Label
+import org.objectweb.asm.Type
+import java.util.Collections.unmodifiableSet
+
+internal fun List<LocalVariableInfo>.isUniqueVariable(): Boolean {
+    val name = first().name
+    val type = first().type
+    return all { it.name == name && it.type == type }
+}
+
+internal data class LocalVariableInfo(val name: String, val index: Int, val labelIndexRange: Pair<Label, Label>, val type: Type) {
+    val isInlineCallMarker = name.startsWith("\$i\$f\$")
+    val inlineMethodName = if (isInlineCallMarker) name.substring(5) else name
```

Comment:
I think it is better to set `inlineMethodName` it to `null` if `isInlineCallMarker` is false.